packages/calcite-components/src/components/list/list.tsx

```diff
@@ -477,6 +486,7 @@ export class List implements InteractiveComponent, LoadableComponent, SortableCo
   }
 
   onDragSort(event: SortableEvent): void {
+    this.parentListEl = this.el.parentElement.closest("calcite-list");
```

Comment:
Since connectedCallback won't be fired, we should update the parentList here once dragging/sorting is done.